api/src/service/domain/document/document.ts

```diff
@@ -21,31 +21,60 @@ export const storedDocumentSchema = Joi.object({
 export interface DocumentReference {
   id: string;
   fileName: string;
-  hash: string;
+  hash?: string;
+  link?: string;
```

Comment:
The hash of the document should not be optional when using the storage service. I see there 2 options:
- let it be optional but check with Joi, that documents on the storage service have the hash. Maybe splitting the interfaces would also help / improve readability
- or (but we have to discuss this) also store the hash of the document which is stored on a remote storage. this would imply downloading the document & computing the hash